src/warning/warning-printer.ts

```diff
@@ -120,7 +121,7 @@ export class WarningPrinter {
   private async _getLinesOfText(
       startLine: number, endLine: number, localPath: string) {
     const contents = await this._options.analyzer.load(localPath);
-    return contents.split('\n').slice(startLine, endLine + 1);
+    return contents.split(EOL).slice(startLine, endLine + 1);
```

Comment:
I can't spot any bugs. It feels a bit iffy to define source ranges by `\n` but to handle both `\r\n` and `\n` here. OTOH, printing a standalone `\r` is a great way to get super weird unexpected console output, so let's try this and see what happens. Worst case I is that we make some off-by-one mistakes when printing squiggles on windows.

Hm, it's probably worth adding a test of printing a warning in a document with `\r\n`s